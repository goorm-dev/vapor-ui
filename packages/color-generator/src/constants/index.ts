export * from './colors';
export * from './thresholds';
