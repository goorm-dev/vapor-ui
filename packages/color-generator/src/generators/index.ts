export * from './system-color-palette';
export * from './brand-color-palette';
export * from './semantic-mapping';
