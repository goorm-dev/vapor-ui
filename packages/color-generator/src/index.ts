/* -------------------------------------------------------------------------------------------------
 * Color Generators
 * -----------------------------------------------------------------------------------------------*/

export * from './generators';

/* -------------------------------------------------------------------------------------------------
 * Constants
 * -----------------------------------------------------------------------------------------------*/

export * from './constants';

/* -------------------------------------------------------------------------------------------------
 * Types
 * -----------------------------------------------------------------------------------------------*/

export type * from './types';

/* -------------------------------------------------------------------------------------------------
 * Utils
 * -----------------------------------------------------------------------------------------------*/

export { getColorLightness } from './utils';
