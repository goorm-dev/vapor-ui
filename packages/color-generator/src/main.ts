import * as fs from 'fs';
import * as path from 'path';
import { fileURLToPath } from 'url';

import { darkThemeCss, lightThemeCss } from './index';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);
const dataDir = path.join(__dirname, 'data');
if (!fs.existsSync(dataDir)) {
    fs.mkdirSync(dataDir, { recursive: true });
}

fs.writeFileSync(path.join(dataDir, 'renewal-vapor-color-light-theme.css'), lightThemeCss);
fs.writeFileSync(path.join(dataDir, 'renewal-vapor-color-dark-theme.css'), darkThemeCss);
