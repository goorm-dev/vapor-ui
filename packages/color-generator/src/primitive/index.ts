
import {
    type ColorGeneratorConfig,
    type ColorPaletteCollection,
    DEFAULT_CONTRAST_RATIOS,
    createBaseColorTokens,
    formatOklchForWeb,
} from '../core';
import { generateThemeTokens } from '../libs/adobe-leonardo';

// ============================================================================
// Primitive Colors Configuration
// ============================================================================

export const DEFAULT_PRIMITIVE_COLORS = {
    red: '#F5535E',
    pink: '#F26394',
    grape: '#CC5DE8',
    violet: '#8662F3',
    blue: '#448EFE',
    cyan: '#1EBAD2',
    green: '#04A37E',
    lime: '#1EBAD2',
    yellow: '#FFC107',
    orange: '#ED670C',
} as const;

// ============================================================================
// Primitive Color Generation Logic
// ============================================================================


export function generateColorPalette(config: ColorGeneratorConfig = {}): ColorPaletteCollection {
    const primitiveColors = config.primitiveColors || DEFAULT_PRIMITIVE_COLORS;
    const contrastRatios = config.contrastRatios || DEFAULT_CONTRAST_RATIOS;


    return {
        base: createBaseColorTokens(formatOklchForWeb),
        light: generateThemeTokens(primitiveColors, contrastRatios, 'light'),
        dark: generateThemeTokens(primitiveColors, contrastRatios, 'dark'),
    };
}
