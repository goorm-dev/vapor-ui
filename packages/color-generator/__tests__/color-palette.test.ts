import { expect, describe, it } from 'vitest';
import { colorPalette } from '../src/index';

describe('Color Generator', () => {
  describe('colorPalette', () => {
    it('should generate consistent color palette collection', () => {
      expect(colorPalette).toMatchSnapshot();
    });

    it('should have required structure', () => {
      expect(colorPalette).toHaveProperty('base');
      expect(colorPalette).toHaveProperty('light');
      expect(colorPalette).toHaveProperty('dark');
      
      expect(colorPalette.base).toHaveProperty('white');
      expect(colorPalette.base).toHaveProperty('black');
      
      expect(colorPalette.light).toHaveProperty('background.canvas');
      expect(colorPalette.dark).toHaveProperty('background.canvas');
    });

    it('should have valid base colors', () => {
      const hexRegex = /^#[0-9A-Fa-f]{6}$/;
      const oklchRegex = /^oklch\(/;
      
      expect(colorPalette.base.white.hex).toMatch(hexRegex);
      expect(colorPalette.base.white.oklch).toMatch(oklchRegex);
      expect(colorPalette.base.black.hex).toMatch(hexRegex);
      expect(colorPalette.base.black.oklch).toMatch(oklchRegex);
    });

    it('should have valid light theme colors', () => {
      const hexRegex = /^#[0-9A-Fa-f]{6}$/;
      const oklchRegex = /^oklch\(/;
      
      Object.values(colorPalette.light).forEach(colorGroup => {
        Object.values(colorGroup).forEach(colorToken => {
          expect(colorToken.hex).toMatch(hexRegex);
          expect(colorToken.oklch).toMatch(oklchRegex);
        });
      });
    });

    it('should have valid dark theme colors', () => {
      const hexRegex = /^#[0-9A-Fa-f]{6}$/;
      const oklchRegex = /^oklch\(/;
      
      Object.values(colorPalette.dark).forEach(colorGroup => {
        Object.values(colorGroup).forEach(colorToken => {
          expect(colorToken.hex).toMatch(hexRegex);
          expect(colorToken.oklch).toMatch(oklchRegex);
        });
      });
    });

    it('should have canvas background tokens', () => {
      expect(colorPalette.light.background.canvas).toBeDefined();
      expect(colorPalette.dark.background.canvas).toBeDefined();
      
      expect(colorPalette.light.background.canvas.hex).toBe('#ffffff');
      expect(colorPalette.dark.background.canvas.hex).toMatch(/^#[0-9a-f]{6}$/i);
      
      // Dark canvas should be different from light canvas
      expect(colorPalette.dark.background.canvas.hex)
        .not.toBe(colorPalette.light.background.canvas.hex);
    });

    it('should have same color categories in both themes', () => {
      const lightCategories = Object.keys(colorPalette.light).sort();
      const darkCategories = Object.keys(colorPalette.dark).sort();
      
      expect(lightCategories).toEqual(darkCategories);
    });

    it('should have same shade counts for each color', () => {
      Object.keys(colorPalette.light).forEach(category => {
        const lightShades = Object.keys(colorPalette.light[category]).sort();
        const darkShades = Object.keys(colorPalette.dark[category]).sort();
        
        expect(lightShades).toEqual(darkShades);
      });
    });

    it('should use 3-digit shade format (050 instead of 50)', () => {
      // Check if any color groups have '050' shade instead of '50'
      Object.keys(colorPalette.light).forEach(category => {
        if (category !== 'background') {
          const shades = Object.keys(colorPalette.light[category]);
          
          // Should have '050' not '50'
          expect(shades).toContain('050');
          expect(shades).not.toContain('50');
        }
      });
    });

    it('should have properly formatted OKLCH values', () => {
      // OKLCH format should be: oklch(L C H) with reasonable decimal precision
      const oklchRegex = /^oklch\([\d.]{1,5}\s+[\d.]{1,5}\s+[\d.]{1,5}\)$/;
      
      // Test base colors
      expect(colorPalette.base.white.oklch).toMatch(oklchRegex);
      expect(colorPalette.base.black.oklch).toMatch(oklchRegex);
      
      // Test theme colors
      Object.values(colorPalette.light).forEach(colorGroup => {
        Object.values(colorGroup).forEach(colorToken => {
          expect(colorToken.oklch).toMatch(oklchRegex);
          
          // Check that decimal places are reasonable (not overly long)
          const match = colorToken.oklch.match(/oklch\(([\d.]+)\s+([\d.]+)\s+([\d.]+)\)/);
          if (match) {
            const [, l, c, h] = match;
            // L and C should have max 3 decimal places
            expect(l.split('.')[1]?.length || 0).toBeLessThanOrEqual(3);
            expect(c.split('.')[1]?.length || 0).toBeLessThanOrEqual(3);
            // H should have max 1 decimal place
            expect(h.split('.')[1]?.length || 0).toBeLessThanOrEqual(1);
          }
        });
      });
    });

    it('should have all expected shade variations', () => {
      const expectedShades = ['050', '100', '200', '300', '400', '500', '600', '700', '800', '900'];
      
      Object.keys(colorPalette.light).forEach(category => {
        if (category !== 'background') {
          const shades = Object.keys(colorPalette.light[category]).sort();
          expect(shades).toEqual(expectedShades);
        }
      });
    });
  });
});