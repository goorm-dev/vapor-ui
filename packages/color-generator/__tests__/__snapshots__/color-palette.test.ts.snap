// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Color Generator > colorPalette > should generate consistent color palette collection 1`] = `
{
  "base": {
    "black": {
      "hex": "#000000",
      "oklch": "oklch(0.000 0.000 0.0)",
    },
    "white": {
      "hex": "#ffffff",
      "oklch": "oklch(1.000 0.000 0.0)",
    },
  },
  "dark": {
    "background": {
      "canvas": {
        "hex": "#232323",
        "oklch": "oklch(0.256 0.000 0.0)",
      },
    },
    "blue": {
      "050": {
        "hex": "#00257a",
        "oklch": "oklch(0.311 0.149 262.5)",
      },
      "100": {
        "hex": "#002f85",
        "oklch": "oklch(0.343 0.152 261.4)",
      },
      "200": {
        "hex": "#09429c",
        "oklch": "oklch(0.406 0.157 260.3)",
      },
      "300": {
        "hex": "#1f5cbe",
        "oklch": "oklch(0.494 0.166 259.8)",
      },
      "400": {
        "hex": "#2969cf",
        "oklch": "oklch(0.538 0.171 259.6)",
      },
      "500": {
        "hex": "#3f87f6",
        "oklch": "oklch(0.635 0.181 259.0)",
      },
      "600": {
        "hex": "#6da8ff",
        "oklch": "oklch(0.729 0.141 258.0)",
      },
      "700": {
        "hex": "#96c1ff",
        "oklch": "oklch(0.803 0.100 257.5)",
      },
      "800": {
        "hex": "#c7deff",
        "oklch": "oklch(0.894 0.052 257.4)",
      },
      "900": {
        "hex": "#f6faff",
        "oklch": "oklch(0.984 0.008 253.9)",
      },
    },
    "cyan": {
      "050": {
        "hex": "#003244",
        "oklch": "oklch(0.297 0.058 228.9)",
      },
      "100": {
        "hex": "#003c4e",
        "oklch": "oklch(0.331 0.063 225.2)",
      },
      "200": {
        "hex": "#004f61",
        "oklch": "oklch(0.395 0.072 219.9)",
      },
      "300": {
        "hex": "#05697e",
        "oklch": "oklch(0.481 0.085 218.2)",
      },
      "400": {
        "hex": "#09768c",
        "oklch": "oklch(0.523 0.091 217.2)",
      },
      "500": {
        "hex": "#1396ae",
        "oklch": "oklch(0.621 0.106 214.8)",
      },
      "600": {
        "hex": "#1db6ce",
        "oklch": "oklch(0.714 0.120 211.9)",
      },
      "700": {
        "hex": "#6fccdd",
        "oklch": "oklch(0.794 0.092 210.8)",
      },
      "800": {
        "hex": "#b8e4ee",
        "oklch": "oklch(0.891 0.048 213.1)",
      },
      "900": {
        "hex": "#f4fbfc",
        "oklch": "oklch(0.983 0.008 207.1)",
      },
    },
    "grape": {
      "050": {
        "hex": "#510065",
        "oklch": "oklch(0.321 0.156 317.5)",
      },
      "100": {
        "hex": "#5f0074",
        "oklch": "oklch(0.357 0.173 318.3)",
      },
      "200": {
        "hex": "#77168d",
        "oklch": "oklch(0.425 0.187 319.0)",
      },
      "300": {
        "hex": "#9532ad",
        "oklch": "oklch(0.514 0.198 319.1)",
      },
      "400": {
        "hex": "#a43ebe",
        "oklch": "oklch(0.558 0.205 318.8)",
      },
      "500": {
        "hex": "#c85ae3",
        "oklch": "oklch(0.658 0.217 319.3)",
      },
      "600": {
        "hex": "#da87ef",
        "oklch": "oklch(0.747 0.168 319.3)",
      },
      "700": {
        "hex": "#e5a9f5",
        "oklch": "oklch(0.816 0.122 319.0)",
      },
      "800": {
        "hex": "#f2d2fb",
        "oklch": "oklch(0.903 0.065 318.8)",
      },
      "900": {
        "hex": "#fdf8fe",
        "oklch": "oklch(0.985 0.009 320.5)",
      },
    },
    "gray": {
      "050": {
        "hex": "#2e2e2e",
        "oklch": "oklch(0.301 0.000 0.0)",
      },
      "100": {
        "hex": "#363636",
        "oklch": "oklch(0.333 0.000 0.0)",
      },
      "200": {
        "hex": "#474747",
        "oklch": "oklch(0.398 0.000 0.0)",
      },
      "300": {
        "hex": "#606060",
        "oklch": "oklch(0.489 0.000 0.0)",
      },
      "400": {
        "hex": "#6c6c6c",
        "oklch": "oklch(0.531 0.000 0.0)",
      },
      "500": {
        "hex": "#898989",
        "oklch": "oklch(0.630 0.000 0.0)",
      },
      "600": {
        "hex": "#a7a7a7",
        "oklch": "oklch(0.728 0.000 0.0)",
      },
      "700": {
        "hex": "#bebebe",
        "oklch": "oklch(0.802 0.000 0.0)",
      },
      "800": {
        "hex": "#dcdcdc",
        "oklch": "oklch(0.894 0.000 0.0)",
      },
      "900": {
        "hex": "#fafafa",
        "oklch": "oklch(0.985 0.000 0.0)",
      },
    },
    "green": {
      "050": {
        "hex": "#00361e",
        "oklch": "oklch(0.293 0.069 157.6)",
      },
      "100": {
        "hex": "#003f26",
        "oklch": "oklch(0.324 0.074 159.5)",
      },
      "200": {
        "hex": "#015237",
        "oklch": "oklch(0.388 0.084 162.7)",
      },
      "300": {
        "hex": "#046e4e",
        "oklch": "oklch(0.477 0.100 164.5)",
      },
      "400": {
        "hex": "#057b5a",
        "oklch": "oklch(0.517 0.106 166.1)",
      },
      "500": {
        "hex": "#059c77",
        "oklch": "oklch(0.616 0.123 168.3)",
      },
      "600": {
        "hex": "#5cb798",
        "oklch": "oklch(0.714 0.100 168.9)",
      },
      "700": {
        "hex": "#8ecbb4",
        "oklch": "oklch(0.794 0.071 169.0)",
      },
      "800": {
        "hex": "#c3e3d7",
        "oklch": "oklch(0.889 0.037 170.8)",
      },
      "900": {
        "hex": "#f5fbf8",
        "oklch": "oklch(0.983 0.007 164.9)",
      },
    },
    "lime": {
      "050": {
        "hex": "#003244",
        "oklch": "oklch(0.297 0.058 228.9)",
      },
      "100": {
        "hex": "#003c4e",
        "oklch": "oklch(0.331 0.063 225.2)",
      },
      "200": {
        "hex": "#004f61",
        "oklch": "oklch(0.395 0.072 219.9)",
      },
      "300": {
        "hex": "#05697e",
        "oklch": "oklch(0.481 0.085 218.2)",
      },
      "400": {
        "hex": "#09768c",
        "oklch": "oklch(0.523 0.091 217.2)",
      },
      "500": {
        "hex": "#1396ae",
        "oklch": "oklch(0.621 0.106 214.8)",
      },
      "600": {
        "hex": "#1db6ce",
        "oklch": "oklch(0.714 0.120 211.9)",
      },
      "700": {
        "hex": "#6fccdd",
        "oklch": "oklch(0.794 0.092 210.8)",
      },
      "800": {
        "hex": "#b8e4ee",
        "oklch": "oklch(0.891 0.048 213.1)",
      },
      "900": {
        "hex": "#f4fbfc",
        "oklch": "oklch(0.983 0.008 207.1)",
      },
    },
    "orange": {
      "050": {
        "hex": "#640000",
        "oklch": "oklch(0.316 0.130 29.2)",
      },
      "100": {
        "hex": "#750000",
        "oklch": "oklch(0.353 0.145 29.2)",
      },
      "200": {
        "hex": "#8c1b02",
        "oklch": "oklch(0.417 0.151 33.1)",
      },
      "300": {
        "hex": "#af3606",
        "oklch": "oklch(0.508 0.165 37.5)",
      },
      "400": {
        "hex": "#bf4308",
        "oklch": "oklch(0.551 0.170 39.9)",
      },
      "500": {
        "hex": "#e5600c",
        "oklch": "oklch(0.649 0.183 44.6)",
      },
      "600": {
        "hex": "#f58a56",
        "oklch": "oklch(0.740 0.147 45.6)",
      },
      "700": {
        "hex": "#fbac88",
        "oklch": "oklch(0.812 0.105 45.3)",
      },
      "800": {
        "hex": "#ffd4c0",
        "oklch": "oklch(0.902 0.056 46.3)",
      },
      "900": {
        "hex": "#fff9f5",
        "oklch": "oklch(0.986 0.008 56.3)",
      },
    },
    "pink": {
      "050": {
        "hex": "#610028",
        "oklch": "oklch(0.316 0.127 5.9)",
      },
      "100": {
        "hex": "#700033",
        "oklch": "oklch(0.351 0.141 3.5)",
      },
      "200": {
        "hex": "#8b1247",
        "oklch": "oklch(0.420 0.157 1.6)",
      },
      "300": {
        "hex": "#ad2f60",
        "oklch": "oklch(0.512 0.166 1.5)",
      },
      "400": {
        "hex": "#bd3c6c",
        "oklch": "oklch(0.555 0.169 1.6)",
      },
      "500": {
        "hex": "#e35889",
        "oklch": "oklch(0.654 0.178 1.4)",
      },
      "600": {
        "hex": "#f881a6",
        "oklch": "oklch(0.746 0.150 1.6)",
      },
      "700": {
        "hex": "#fea5be",
        "oklch": "oklch(0.817 0.109 1.5)",
      },
      "800": {
        "hex": "#ffd0dd",
        "oklch": "oklch(0.902 0.055 359.8)",
      },
      "900": {
        "hex": "#fff8fa",
        "oklch": "oklch(0.985 0.008 357.8)",
      },
    },
    "red": {
      "050": {
        "hex": "#630000",
        "oklch": "oklch(0.314 0.129 29.2)",
      },
      "100": {
        "hex": "#730006",
        "oklch": "oklch(0.349 0.143 27.3)",
      },
      "200": {
        "hex": "#900f1d",
        "oklch": "oklch(0.419 0.160 23.6)",
      },
      "300": {
        "hex": "#b52b34",
        "oklch": "oklch(0.512 0.174 22.7)",
      },
      "400": {
        "hex": "#c63640",
        "oklch": "oklch(0.554 0.180 21.8)",
      },
      "500": {
        "hex": "#f0505b",
        "oklch": "oklch(0.654 0.196 20.6)",
      },
      "600": {
        "hex": "#fd8283",
        "oklch": "oklch(0.745 0.150 20.8)",
      },
      "700": {
        "hex": "#ffa7a6",
        "oklch": "oklch(0.815 0.105 20.3)",
      },
      "800": {
        "hex": "#ffd2d0",
        "oklch": "oklch(0.902 0.051 21.1)",
      },
      "900": {
        "hex": "#fff8f8",
        "oklch": "oklch(0.985 0.007 17.3)",
      },
    },
    "violet": {
      "050": {
        "hex": "#390684",
        "oklch": "oklch(0.322 0.176 289.8)",
      },
      "100": {
        "hex": "#421690",
        "oklch": "oklch(0.357 0.179 290.4)",
      },
      "200": {
        "hex": "#522ba8",
        "oklch": "oklch(0.420 0.186 290.3)",
      },
      "300": {
        "hex": "#6a45cc",
        "oklch": "oklch(0.509 0.198 290.2)",
      },
      "400": {
        "hex": "#7652dd",
        "oklch": "oklch(0.553 0.202 290.1)",
      },
      "500": {
        "hex": "#9074f6",
        "oklch": "oklch(0.647 0.187 290.1)",
      },
      "600": {
        "hex": "#aa99fb",
        "oklch": "oklch(0.737 0.140 290.5)",
      },
      "700": {
        "hex": "#c0b6fe",
        "oklch": "oklch(0.810 0.101 290.3)",
      },
      "800": {
        "hex": "#ddd9ff",
        "oklch": "oklch(0.900 0.052 290.1)",
      },
      "900": {
        "hex": "#faf9ff",
        "oklch": "oklch(0.985 0.008 293.9)",
      },
    },
    "yellow": {
      "050": {
        "hex": "#4d2100",
        "oklch": "oklch(0.306 0.079 51.7)",
      },
      "100": {
        "hex": "#572a00",
        "oklch": "oklch(0.339 0.083 56.1)",
      },
      "200": {
        "hex": "#6b3c00",
        "oklch": "oklch(0.405 0.092 63.3)",
      },
      "300": {
        "hex": "#875500",
        "oklch": "oklch(0.494 0.107 70.5)",
      },
      "400": {
        "hex": "#966101",
        "oklch": "oklch(0.537 0.115 72.2)",
      },
      "500": {
        "hex": "#b87e01",
        "oklch": "oklch(0.636 0.133 76.4)",
      },
      "600": {
        "hex": "#d99b02",
        "oklch": "oklch(0.730 0.151 79.8)",
      },
      "700": {
        "hex": "#f3b405",
        "oklch": "oklch(0.807 0.165 83.0)",
      },
      "800": {
        "hex": "#ffd782",
        "oklch": "oklch(0.895 0.113 85.0)",
      },
      "900": {
        "hex": "#fff9ee",
        "oklch": "oklch(0.984 0.016 82.8)",
      },
    },
  },
  "light": {
    "background": {
      "canvas": {
        "hex": "#ffffff",
        "oklch": "oklch(1.000 0.000 0.0)",
      },
    },
    "blue": {
      "050": {
        "hex": "#e3efff",
        "oklch": "oklch(0.948 0.025 255.6)",
      },
      "100": {
        "hex": "#cee3ff",
        "oklch": "oklch(0.909 0.045 255.6)",
      },
      "200": {
        "hex": "#a4c9ff",
        "oklch": "oklch(0.828 0.086 257.7)",
      },
      "300": {
        "hex": "#69a4ff",
        "oklch": "oklch(0.718 0.147 258.8)",
      },
      "400": {
        "hex": "#4e94ff",
        "oklch": "oklch(0.673 0.174 258.7)",
      },
      "500": {
        "hex": "#3174dc",
        "oklch": "oklch(0.573 0.173 259.2)",
      },
      "600": {
        "hex": "#1d5aba",
        "oklch": "oklch(0.486 0.164 259.6)",
      },
      "700": {
        "hex": "#0e47a3",
        "oklch": "oklch(0.424 0.159 260.3)",
      },
      "800": {
        "hex": "#003288",
        "oklch": "oklch(0.352 0.152 261.0)",
      },
      "900": {
        "hex": "#001b71",
        "oklch": "oklch(0.282 0.149 263.4)",
      },
    },
    "cyan": {
      "050": {
        "hex": "#ddf2f7",
        "oklch": "oklch(0.947 0.023 213.6)",
      },
      "100": {
        "hex": "#c2e8f0",
        "oklch": "oklch(0.906 0.041 211.9)",
      },
      "200": {
        "hex": "#84d2e2",
        "oklch": "oklch(0.819 0.080 212.1)",
      },
      "300": {
        "hex": "#1cb3cb",
        "oklch": "oklch(0.706 0.119 212.1)",
      },
      "400": {
        "hex": "#17a3ba",
        "oklch": "oklch(0.659 0.111 212.8)",
      },
      "500": {
        "hex": "#0d8298",
        "oklch": "oklch(0.559 0.097 215.6)",
      },
      "600": {
        "hex": "#04677b",
        "oklch": "oklch(0.474 0.084 217.5)",
      },
      "700": {
        "hex": "#005468",
        "oklch": "oklch(0.413 0.076 221.1)",
      },
      "800": {
        "hex": "#003f51",
        "oklch": "oklch(0.342 0.064 224.2)",
      },
      "900": {
        "hex": "#002a3c",
        "oklch": "oklch(0.268 0.055 232.4)",
      },
    },
    "grape": {
      "050": {
        "hex": "#f9e9fd",
        "oklch": "oklch(0.953 0.031 319.5)",
      },
      "100": {
        "hex": "#f4d8fb",
        "oklch": "oklch(0.916 0.056 319.5)",
      },
      "200": {
        "hex": "#e9b4f7",
        "oklch": "oklch(0.840 0.107 319.1)",
      },
      "300": {
        "hex": "#d883ef",
        "oklch": "oklch(0.739 0.173 318.8)",
      },
      "400": {
        "hex": "#d06bea",
        "oklch": "oklch(0.693 0.202 319.1)",
      },
      "500": {
        "hex": "#b148cb",
        "oklch": "oklch(0.594 0.209 319.1)",
      },
      "600": {
        "hex": "#922fab",
        "oklch": "oklch(0.506 0.198 318.7)",
      },
      "700": {
        "hex": "#7d1c94",
        "oklch": "oklch(0.444 0.190 318.8)",
      },
      "800": {
        "hex": "#640079",
        "oklch": "oklch(0.369 0.178 318.7)",
      },
      "900": {
        "hex": "#450059",
        "oklch": "oklch(0.290 0.142 316.1)",
      },
    },
    "gray": {
      "050": {
        "hex": "#efefef",
        "oklch": "oklch(0.952 0.000 0.0)",
      },
      "100": {
        "hex": "#e1e1e1",
        "oklch": "oklch(0.910 0.000 0.0)",
      },
      "200": {
        "hex": "#c6c6c6",
        "oklch": "oklch(0.827 0.000 0.0)",
      },
      "300": {
        "hex": "#a3a3a3",
        "oklch": "oklch(0.715 0.000 0.0)",
      },
      "400": {
        "hex": "#959595",
        "oklch": "oklch(0.670 0.000 0.0)",
      },
      "500": {
        "hex": "#767676",
        "oklch": "oklch(0.566 0.000 0.0)",
      },
      "600": {
        "hex": "#5d5d5d",
        "oklch": "oklch(0.478 0.000 0.0)",
      },
      "700": {
        "hex": "#4c4c4c",
        "oklch": "oklch(0.417 0.000 0.0)",
      },
      "800": {
        "hex": "#393939",
        "oklch": "oklch(0.345 0.000 0.0)",
      },
      "900": {
        "hex": "#262626",
        "oklch": "oklch(0.269 0.000 0.0)",
      },
    },
    "green": {
      "050": {
        "hex": "#e3f2ec",
        "oklch": "oklch(0.948 0.018 170.1)",
      },
      "100": {
        "hex": "#cde8dd",
        "oklch": "oklch(0.908 0.032 169.2)",
      },
      "200": {
        "hex": "#9cd2bd",
        "oklch": "oklch(0.820 0.063 168.8)",
      },
      "300": {
        "hex": "#55b495",
        "oklch": "oklch(0.703 0.102 169.4)",
      },
      "400": {
        "hex": "#28a784",
        "oklch": "oklch(0.653 0.119 169.5)",
      },
      "500": {
        "hex": "#058765",
        "oklch": "oklch(0.554 0.112 167.3)",
      },
      "600": {
        "hex": "#046b4c",
        "oklch": "oklch(0.468 0.098 164.7)",
      },
      "700": {
        "hex": "#01583c",
        "oklch": "oklch(0.408 0.087 163.3)",
      },
      "800": {
        "hex": "#004229",
        "oklch": "oklch(0.335 0.075 160.4)",
      },
      "900": {
        "hex": "#002d16",
        "oklch": "oklch(0.261 0.064 155.3)",
      },
    },
    "lime": {
      "050": {
        "hex": "#ddf2f7",
        "oklch": "oklch(0.947 0.023 213.6)",
      },
      "100": {
        "hex": "#c2e8f0",
        "oklch": "oklch(0.906 0.041 211.9)",
      },
      "200": {
        "hex": "#84d2e2",
        "oklch": "oklch(0.819 0.080 212.1)",
      },
      "300": {
        "hex": "#1cb3cb",
        "oklch": "oklch(0.706 0.119 212.1)",
      },
      "400": {
        "hex": "#17a3ba",
        "oklch": "oklch(0.659 0.111 212.8)",
      },
      "500": {
        "hex": "#0d8298",
        "oklch": "oklch(0.559 0.097 215.6)",
      },
      "600": {
        "hex": "#04677b",
        "oklch": "oklch(0.474 0.084 217.5)",
      },
      "700": {
        "hex": "#005468",
        "oklch": "oklch(0.413 0.076 221.1)",
      },
      "800": {
        "hex": "#003f51",
        "oklch": "oklch(0.342 0.064 224.2)",
      },
      "900": {
        "hex": "#002a3c",
        "oklch": "oklch(0.268 0.055 232.4)",
      },
    },
    "orange": {
      "050": {
        "hex": "#ffeae0",
        "oklch": "oklch(0.951 0.027 47.2)",
      },
      "100": {
        "hex": "#ffd9c8",
        "oklch": "oklch(0.913 0.048 45.5)",
      },
      "200": {
        "hex": "#fcb797",
        "oklch": "oklch(0.836 0.092 45.9)",
      },
      "300": {
        "hex": "#f4864f",
        "oklch": "oklch(0.731 0.151 45.9)",
      },
      "400": {
        "hex": "#ef6f25",
        "oklch": "oklch(0.685 0.177 45.9)",
      },
      "500": {
        "hex": "#cd4d0a",
        "oklch": "oklch(0.586 0.175 41.4)",
      },
      "600": {
        "hex": "#ab3406",
        "oklch": "oklch(0.499 0.162 37.3)",
      },
      "700": {
        "hex": "#932103",
        "oklch": "oklch(0.436 0.154 34.1)",
      },
      "800": {
        "hex": "#780600",
        "oklch": "oklch(0.363 0.145 30.2)",
      },
      "900": {
        "hex": "#570000",
        "oklch": "oklch(0.287 0.118 29.2)",
      },
    },
    "pink": {
      "050": {
        "hex": "#ffeaf0",
        "oklch": "oklch(0.956 0.024 358.3)",
      },
      "100": {
        "hex": "#ffd8e2",
        "oklch": "oklch(0.918 0.045 0.8)",
      },
      "200": {
        "hex": "#ffb1c6",
        "oklch": "oklch(0.840 0.094 1.7)",
      },
      "300": {
        "hex": "#f77ca3",
        "oklch": "oklch(0.737 0.155 1.5)",
      },
      "400": {
        "hex": "#f26394",
        "oklch": "oklch(0.692 0.181 1.7)",
      },
      "500": {
        "hex": "#ca4676",
        "oklch": "oklch(0.590 0.172 1.5)",
      },
      "600": {
        "hex": "#aa2d5d",
        "oklch": "oklch(0.504 0.165 1.9)",
      },
      "700": {
        "hex": "#92194d",
        "oklch": "oklch(0.440 0.159 1.1)",
      },
      "800": {
        "hex": "#760037",
        "oklch": "oklch(0.365 0.147 2.9)",
      },
      "900": {
        "hex": "#54001e",
        "oklch": "oklch(0.285 0.114 9.1)",
      },
    },
    "red": {
      "050": {
        "hex": "#ffeae9",
        "oklch": "oklch(0.954 0.023 20.9)",
      },
      "100": {
        "hex": "#ffd8d7",
        "oklch": "oklch(0.915 0.044 19.7)",
      },
      "200": {
        "hex": "#ffb3b2",
        "oklch": "oklch(0.838 0.089 20.0)",
      },
      "300": {
        "hex": "#fc7d7f",
        "oklch": "oklch(0.736 0.155 20.6)",
      },
      "400": {
        "hex": "#f8636a",
        "oklch": "oklch(0.691 0.183 20.5)",
      },
      "500": {
        "hex": "#d54049",
        "oklch": "oklch(0.590 0.185 21.7)",
      },
      "600": {
        "hex": "#b12932",
        "oklch": "oklch(0.503 0.172 22.8)",
      },
      "700": {
        "hex": "#971622",
        "oklch": "oklch(0.437 0.162 23.4)",
      },
      "800": {
        "hex": "#79000c",
        "oklch": "oklch(0.363 0.147 25.9)",
      },
      "900": {
        "hex": "#560000",
        "oklch": "oklch(0.285 0.117 29.2)",
      },
    },
    "violet": {
      "050": {
        "hex": "#efedff",
        "oklch": "oklch(0.953 0.024 291.0)",
      },
      "100": {
        "hex": "#e1deff",
        "oklch": "oklch(0.913 0.045 289.7)",
      },
      "200": {
        "hex": "#c8bffe",
        "oklch": "oklch(0.834 0.088 290.9)",
      },
      "300": {
        "hex": "#a796fa",
        "oklch": "oklch(0.729 0.143 290.2)",
      },
      "400": {
        "hex": "#9a83f8",
        "oklch": "oklch(0.683 0.168 290.3)",
      },
      "500": {
        "hex": "#805ceb",
        "oklch": "oklch(0.587 0.206 290.2)",
      },
      "600": {
        "hex": "#6843c8",
        "oklch": "oklch(0.502 0.195 290.4)",
      },
      "700": {
        "hex": "#5731b0",
        "oklch": "oklch(0.440 0.188 290.1)",
      },
      "800": {
        "hex": "#441a94",
        "oklch": "oklch(0.367 0.180 290.0)",
      },
      "900": {
        "hex": "#300077",
        "oklch": "oklch(0.292 0.167 288.5)",
      },
    },
    "yellow": {
      "050": {
        "hex": "#ffecc3",
        "oklch": "oklch(0.948 0.057 86.1)",
      },
      "100": {
        "hex": "#ffdd95",
        "oklch": "oklch(0.910 0.098 85.3)",
      },
      "200": {
        "hex": "#fbbd06",
        "oklch": "oklch(0.832 0.170 84.5)",
      },
      "300": {
        "hex": "#d69902",
        "oklch": "oklch(0.723 0.149 80.0)",
      },
      "400": {
        "hex": "#c58a01",
        "oklch": "oklch(0.675 0.140 78.3)",
      },
      "500": {
        "hex": "#a26c01",
        "oklch": "oklch(0.574 0.121 74.5)",
      },
      "600": {
        "hex": "#855300",
        "oklch": "oklch(0.487 0.106 69.9)",
      },
      "700": {
        "hex": "#714100",
        "oklch": "oklch(0.424 0.095 64.7)",
      },
      "800": {
        "hex": "#5b2d00",
        "oklch": "oklch(0.352 0.085 57.0)",
      },
      "900": {
        "hex": "#461900",
        "oklch": "oklch(0.279 0.078 46.4)",
      },
    },
  },
}
`;
